test/widgets/compose_box_test.dart

```diff
@@ -505,5 +505,151 @@ void main() {
+        (ChannelPostPolicy.moderators,     UserRole.administrator, true),
+        (ChannelPostPolicy.moderators,     UserRole.owner,         true),
+        (ChannelPostPolicy.administrators, UserRole.unknown,       true),
+        (ChannelPostPolicy.administrators, UserRole.guest,         false),
+        (ChannelPostPolicy.administrators, UserRole.member,        false),
+        (ChannelPostPolicy.administrators, UserRole.moderator,     false),
+        (ChannelPostPolicy.administrators, UserRole.administrator, true),
+        (ChannelPostPolicy.administrators, UserRole.owner,         true),
+      ];
+
+      for (final (ChannelPostPolicy policy, UserRole role, bool canPost) in testCases) {
+        testWidgets('"${role.name}" user ${canPost ? 'can' : "can't"} post in channel with "${policy.name}" policy', (tester) async {
+          final selfUser = eg.user(role: role);
+          await prepareComposeBox(tester,
+            narrow: const ChannelNarrow(1),
+            selfUser: selfUser,
+            streams: [eg.stream(streamId: 1, channelPostPolicy: policy)]);
+          checkComposeBox(isShown: canPost);
+        });
+
+        testWidgets('"${role.name}" user ${canPost ? 'can' : "can't"} post in topic with "${policy.name}" channel policy', (tester) async {
+          final selfUser = eg.user(role: role);
+          await prepareComposeBox(tester,
+            narrow: const TopicNarrow(1, 'topic'),
+            selfUser: selfUser,
+            streams: [eg.stream(streamId: 1, channelPostPolicy: policy)]);
+          checkComposeBox(isShown: canPost);
+        });
+      }
+
+      group('only "full member" user can post in channel with "fullMembers" policy', () {
+        testWidgets('"full member" -> can post in channel', (tester) async {
+          final selfUser = eg.user(role: UserRole.member,
+            dateJoined: DateTime.now().subtract(const Duration(days: 30)).toIso8601String());
+          await prepareComposeBox(tester,
+            narrow: const ChannelNarrow(1),
+            selfUser: selfUser,
+            realmWaitingPeriodThreshold: 30,
+            streams: [eg.stream(streamId: 1, channelPostPolicy: ChannelPostPolicy.fullMembers)]);
+          checkComposeBox(isShown: true);
+        });
+
+        testWidgets('not a "full member" -> cannot post in channel', (tester) async {
+          final selfUser = eg.user(role: UserRole.member,
+            dateJoined: DateTime.now().subtract(const Duration(days: 29)).toIso8601String());
+          await prepareComposeBox(tester,
+            narrow: const ChannelNarrow(1),
+            selfUser: selfUser,
+            realmWaitingPeriodThreshold: 30,
+            streams: [eg.stream(streamId: 1, channelPostPolicy: ChannelPostPolicy.fullMembers)]);
+          checkComposeBox(isShown: false);
+        });
+      });
+
+      Future<void> changeUserRole(WidgetTester tester,
+          {required User user, required UserRole role}) async {
```

Comment:
nit: Formatting
```suggestion
      Future<void> changeUserRole(WidgetTester tester, {
        required User user,
        required UserRole role,
      }) async {
```
We usually do this for named parameters when wrapping them.